lib/Mail/Pyzor/Client.pm

```diff
@@ -348,15 +348,21 @@ sub _send_packet {
 
 sub _get_connection_or_die {
     my ($self) = @_;
-    $self->{'_sock'} ||= IO::Socket::INET->new(
+
+    # clear the socket if the PID changes
+    if ( defined $self->{'_sock_pid'} && $self->{'_sock_pid'} != $$ ) {
+        undef $self->{'_sock_pid'};
+        undef $self->{'_sock'};
+    }
+
+    $self->{'_sock_pid'} ||= $$;
+    $self->{'_sock'}     ||= IO::Socket::INET->new(
         'PeerHost' => $self->{'_server_host'},
         'PeerPort' => $self->{'_server_port'},
         'Proto'    => 'udp'
-    );
-
-    if ( !$self->{'_sock'} ) {
+    ) or
         die "Cannot connect to $self->{'_server_host'}:$self->{'_server_port'}: $@ $!";
```

Comment:
Does this pass Perl::Tidy? It seems a bit long-winded for a single statement.

It also may run afoul of Perl::Critic’s rule against different-precedence operators in the same statement.

Reply:
this is now tidy